### ISSUE TITLE
Linking of native static library fails and the resulting package is including the native reference

### ISSUE THREAD

**Issue by rmarinho** [xamarin/xamarin-macios#18751]
---
**Issue moved from dotnet/maui#16764**
- Please respond to @tentom.

---

_From @tentom on Tuesday, August 15, 2023 4:19:20 PM_

### Description

When binding a native library the library is not statically linked to the executable.  When the this package is consumed by the MAUI project the library is still not linked in to the iOS app  so when packaged into a .ipa file the library is still not linked in and Apple rejects the app because of a the library being dynamic linked and its not a framework. 

When building and packing the reproduction project (libmds.a is in Git LFS) the file libmds.a is added to the StaticLinkingNotWorking.1.0.0.nupkg under lib -> net7.0-ios16.1 -> StaticLinkingNotWorking.resources -> libmds.a 

What I expected was for it to be linked into the lib ->  net7.0-ios16.1 -> StaticLinkingNotWorking.dll file as it was when using the Xamarin binding project. 

But whatever I do to try to statically link libmds.a it seems like the project insists on making it dynamically linked and add libmds.a as it was before.  

I hope this repository is the correct place to report this issue, if not please point me to the correct place. 

I can consume the package and get a running application locally on a iOS device but this issue causes uploads to AppStore to fail. The exact fail is this one: 

`Asset validation failed (90171) Invalid bundle structure. The “testApp.app/libmds.a” binary file is not permitted. Your app cannot contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. For details, visit: https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle `

(I changed the name of the app to testApp)

### Steps to Reproduce

1. Create a iOS binding project
2. Add native static library (.a) file
3. Build the project
4. pack the project

Expected: 
libmds.a statically linked into the StaticLinkingNotWorking.dll file 

Actual
libmds.a added to the StaticLinkingNotWorking.resources folder and not linked in to the dll file.

### Link to public reproduction project repository

https://github.com/tentom/staticLinkingNotWorking

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Did you find any workaround?

No known workarounds found.

**Comment by rmarinho** [comment]
---
**Issue moved from dotnet/maui#16764**
- Please respond to @samhouts.

---

_From @samhouts on Tuesday, August 15, 2023 8:24:31 PM_

@rolfbjarne Suggestions?

**Comment by rmarinho** [comment]
---
**Issue moved from dotnet/maui#16764**
- Please respond to @drasticactions.

---

_From @drasticactions on Wednesday, August 16, 2023 7:03:10 AM_

This should be transferred to http://github.com/xamarin/xamarin-macios, unless I'm missing some context this isn't related to the MAUI UI project.

**Comment by rmarinho** [comment]
---
**Issue moved from dotnet/maui#16764**
- Please respond to @msftbot[bot].

---

_From @msftbot[bot] on Wednesday, August 16, 2023 8:25:06 AM_

Hi @tentom. We have added the "s/needs-info" label to this issue, which indicates that we have an open question for you before we can take further action. This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

**Comment by rmarinho** [comment]
---
**Issue moved from dotnet/maui#16764**
- Please respond to @tentom.

---

_From @tentom on Wednesday, August 16, 2023 9:44:05 AM_

@rolfbjarne  
Hi, Im not consuming this package directly in a app project, but it goes into a another package that has one android binding (as a NuGet Package) and a iOS binding (as a NuGet Package) as dependencies. This project is in turn made into a NuGet Package and consumed in an MAUI app project. I assume that the .a file should continue to exist in this use case and be removed when consumed in the final executable (app) project in the release configuration,  is this correct? 

I'm not quite sure I understand what you mean by `They should be automatically picked up correctly`.  When I consume the library in my cross platform project and then consume the cross platform package in a MAUI application I can call the native code and everything works as expected and the libmds.a file is still in the generated IPA package. The issue first arises when uploading to the Apple AppStore. So it seems like it is picked up correctly but the final stage there is something missing before the upload to Apple.

1. Should it work like visioned in the first paragraph?
2. Do I have the right understanding of what `They should be automatically picked up correctly` mean? 
3. I will  update the test project with api definitions and consume it in a maui app project and see if the generated IPA file still contains the libmds.a file.

**Comment by rmarinho** [comment]
---
**Issue moved from dotnet/maui#16764**
- Please respond to @rolfbjarne.

---

_From @rolfbjarne on Wednesday, August 16, 2023 11:04:50 AM_

> be removed when consumed in the final executable (app) project in the release configuration, is this correct?

Correct.

> I'm not quite sure I understand what you mean by `They should be automatically picked up correctly`.

I meant that we should automatically do the right thing (link with the .a file, but not include it in the app bundle).

If your app is rejected by the App Store, then something is wrong somewhere.

I might be able to figure out what's happening from a [binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs), but a test project would be better.

**Comment by tentom** [comment]
I am sorry for not responding in a timely manner. 

We do not know exactly what happened or what is the root cause. But here are our findings: 

In the linked GitHub repository there are now one binding project and one app project. There are also a nugget package in the root folder.

When building the app project with the NuGet Package the output of the App project includes the .a file in the generated app, it is not included when building the binding project and including the output form that into the app project.

This is what we have figured out and since the binding project now generated a package without the .a file we have not investigated further, but maybe you see something relating causing the NuGet and the generated nuGet from the binding Project to behave differently when consumed by the app project.

**Comment by borrmann** [comment]
I am also running into this problem. I migrated my app from .NET 7 to .NET 8 and now use CLI to publish my app.

`dotnet publish -f net8.0-ios -c Release -p:ArchiveOnBuild=true -p:CodesignKey="Apple Distribution: XXXXXXXXXX" -p:CodesignProvision="XXXXXXX" -p:MtouchLink=Full -p:MtouchUseLlvm=true  -p:LinkMode=Full`

I have a reference to a library project that has the following proj file:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.2" />
    <PackageReference Include="SQLiteNetExtensions" Version="2.1.0" />
    <PackageReference Include="Stripe.net" Version="43.11.0" />
    <PackageReference Include="TimePeriodLibrary.NET" Version="2.1.5" />
  </ItemGroup>
</Project>
```



When I upload the App to the AppStore I get an error "The XXXX.app/libe_sqlite3.dylib" binary file is not permitted. Same happens with linker set to SdkOnly or None

**Comment by borrmann** [comment]
I don't have any of those packages directly referenced in my MAUI project. I also tried to include the library project directly as a dll instead a project reference, which did not change anything either.

**Comment by rolfbjarne** [comment]
@borrmann please file a new issue and attach a binary build log (https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) and we'll have a look.
